### PR TITLE
fix(plugin): use relative path for css/js imports

### PR DESCRIPTION
**Pull request by tatimblin**
Support subdirectory reverse proxies by using a relative to root url when referencing css and js assets. This item does not handle importing assets at the correct path. That can be configured with the [base](https://vitejs.dev/config/shared-options.html#base) property in the vite config.

To setup a reverse proxy subdirectory the following must be completed:

1. Setup redirects
  a. [domain.com/subdirectory/*](http://domain.com/subdirectory/*) --> [pagescdn.net](http://pagescdn.net/)
  b. [domain.com/assets/*](http://domain.com/assets/*) --> [pagescdn.net/assets/*](http://pagescdn.net/assets/*)
  c. [domain.com/subdirectory](http://domain.com/subdirectory) --> [pagescdn.net/index.html](http://pagescdn.net/index.html)

2. Set base in vite.config.js
  a. base: "../",

3. Use new pages.js version **(THIS PR)**
  a. "@yext/pages": "https://github.com/yext/pages/raw/dev/relativePrefixToRoot/packages/pages/yext-pages-1.0.0-beta.161.tgz",

4. Update paths in starter-site
  a. remove src/ from tailwind references i.e. bg-[url(assets/images/tacos-1.avif)]
  b. move favicon from public/ to assets/

The simplification I mentioned in our call (Luc) was I believe if we got the entire site to serve on [pagescdn.net/subdirectory](http://pagescdn.net/subdirectory) (I’ve only been able to get html pages on the subdirectory), we could remove steps 1b, 2, and 4. Regardless the update for step 3 is required so I can move forward with my PR, and the simplification (nesting the entire site) can be done in a separate item.

J=SLAP-2535
test=manual

compared [yext hosted domain](https://devrelativeprefixtopath-brightly--radical--smelt-pgsdemo-com.preview.pagescdn.com/ny/loc6-chelsea) to [reverse proxy subdirectory domain](https://guileless-eclair-d4f834.netlify.app/subdirectory/ny/loc6-chelsea)

### CODE REVIEW

**Comment by oshi97** [comment]
Some small typos in commit msg `css an js` `That can configured with`

File: packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts [hunk 1]
```diff
@@ -43,17 +43,25 @@ export const reactWrapper = <T extends TemplateRenderProps>(
             new Set()
           )
         )
-          .map((f) => `<link rel="stylesheet" href="/${f}"/>`)
+          .map(
+            (f) =>
+              `<link rel="stylesheet" href="${
+                props.relativePrefixToRoot + f
```
[the remark this answers is not in the document]
Comment (tatimblin) [reply packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts:49]: It and `path` get manually set on every templates `TemplateProps`

**Review by oshi97 — DISMISSED** [review]

**Comment by tatimblin** [comment]
Converted back to a draft because demoing the change in a starter breaks when I remove the binary

File: packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tatimblin) [reply packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts:49]: It is in an extension of that type called `TemplateRenderProps`

[the remark this answers is not in the document]
Comment (tatimblin) [reply packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts:49]: I meant to say it is already there https://github.com/yext/pages/blob/876f48685d270add5e921be16f56565e86ec55a7/packages/pages/src/common/src/template/types.ts#L196

[the remark this answers is not in the document]
Comment (oshi97) [reply packages/pages/src/vite-plugin/build/buildStart/rendering/wrapper.ts:49]: Ooh gotcha

**Review by oshi97 — APPROVED** [review]

**Pull request merged**